### PR TITLE
chore: add linuxstatic-x86 and macos-arm64 binaries

### PR DESCRIPTION
**Pull request by umireon**
Used by https://github.com/Homebrew/homebrew-core/pull/82892

### CODE REVIEW

**Comment by umireon** [comment]
`@zkochan/pkg` is too old to build node14-macos-arm64.

**Comment by zkochan** [comment]
I can rebase it from `vercel/pkg` if it helps

**Comment by umireon** [comment]
@zkochan Thanks! I really need the up to date `@zkochan/pkg` 😭

**Comment by umireon** [comment]
@zkochan Do you know why this M1 build fails with the content of this PR?

```
packages/beta preinstall$ node setup.js
. prepare$ pnpm --dir=verdaccio install && pnpm --dir=fixtures run prepareFixtures && husky install
packages/beta preinstall: internal/modules/cjs/loader.js:892
packages/beta preinstall:   throw err;
packages/beta preinstall:   ^
packages/beta preinstall: Error: Cannot find module '@pnpm/macos-arm64/package.json'
packages/beta preinstall: Require stack:
packages/beta preinstall: - /private/tmp/pnpm-20210810-36505-1le4rl7/pnpm-6.12.1/packages/beta/setup.js
packages/beta preinstall:     at Function.Module._resolveFilename (internal/modules/cjs/loader.js:889:15)
packages/beta preinstall:     at Function.resolve (internal/modules/cjs/helpers.js:98:19)
packages/beta preinstall:     at Object.<anonymous> (/private/tmp/pnpm-20210810-36505-1le4rl7/pnpm-6.12.1/packages/beta/setup.js:12:25)
packages/beta preinstall:     at Module._compile (internal/modules/cjs/loader.js:1072:14)
packages/beta preinstall:     at Object.Module._extensions..js (internal/modules/cjs/loader.js:1101:10)
packages/beta preinstall:     at Module.load (internal/modules/cjs/loader.js:937:32)
packages/beta preinstall:     at Function.Module._load (internal/modules/cjs/loader.js:778:12)
packages/beta preinstall:     at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:76:12)
packages/beta preinstall:     at internal/main/run_main_module.js:17:47 {
packages/beta preinstall:   code: 'MODULE_NOT_FOUND',
packages/beta preinstall:   requireStack: [
packages/beta preinstall:     '/private/tmp/pnpm-20210810-36505-1le4rl7/pnpm-6.12.1/packages/beta/setup.js'
packages/beta preinstall:   ]
packages/beta preinstall: }
packages/beta preinstall: Failed
 ERROR  Command failed with exit code 1.
```

https://github.com/Homebrew/homebrew-core/runs/3288942057

**Comment by zkochan** [comment]
@umireon I guess your new artifact packages should be added here: https://github.com/pnpm/pnpm/blob/2f72e17193f949c7b33ab0435c543636fe64e794/packages/beta/package.json#L11-L15

**Comment by umireon** [comment]
@zkochan Absolutely insightful✌️

**Comment by umireon** [comment]
[The latest node-v14.17.2-macos-x64 binary of vercel/pkg-fetch](https://github.com/vercel/pkg-fetch/releases/download/v3.2/node-v14.17.2-macos-x64) is broken.

## Prerequisites

```
curl -fsSL -o node https://github.com/vercel/pkg-fetch/releases/download/v3.2/node-v14.17.2-macos-x64
chmod a+x node
curl -O https://nodejs.org/download/release/v16.6.1/node-v16.6.1-darwin-x64.tar.gz
```

## main.js

```
var fs = require('fs');
var zlib = require('zlib');

var gzipContent = fs.readFileSync('./node-v16.6.1-darwin-x64.tar.gz');
zlib.gunzip(gzipContent, function (err, binary) {
    console.log(err)
    console.log(binary.length)
});
```

## The error

```
% ./node ./main.js
Error: incorrect data check
    at Zlib.zlibOnError [as onerror] (zlib.js:188:17) {
  errno: -3,
  code: 'Z_DATA_ERROR'
}
/Users/umireon/git/vercelpkg-test/main.js:7
    console.log(binary.length)
                       ^

TypeError: Cannot read property 'length' of undefined
    at Gunzip.cb (/Users/umireon/git/vercelpkg-test/main.js:7:24)
    at Gunzip.zlibBufferOnError (zlib.js:144:8)
    at Gunzip.emit (events.js:375:28)
    at emitErrorNT (internal/streams/destroy.js:106:8)
    at emitErrorCloseNT (internal/streams/destroy.js:74:3)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
```

## Expected

```
% node ./main.js 
null
98787840
```

**Comment by umireon** [comment]
Reported in https://github.com/vercel/pkg-fetch/issues/208.

File: packages/pnpm/package.json [hunk 1]
```diff
@@ -93,6 +92,7 @@
     "p-defer": "^3.0.0",
     "path-exists": "^4.0.0",
     "path-name": "^1.0.0",
+    "pkg": "5.2.1",
```
Comment (zkochan) [inline packages/pnpm/package.json]: it won't work properly with plain pkg. I had to do some changes in the fork to make it work with pnpm.

Comment (umireon) [reply packages/pnpm/package.json]: Sure. I wait for the fork to be updated.

Comment (zkochan) [reply packages/pnpm/package.json]: :ship: `@zkochan/pkg@0.0.0-2`

Comment (umireon) [reply packages/pnpm/package.json]: Thanks!

**Comment by umireon** [comment]
@zkochan node14-macos-x64 of `pkg@5.3.1` is broken (https://github.com/vercel/pkg-fetch/issues/208). Please provide one based on `pkg@5.2.1` and `pkg-fetch@3.1.1`.

**Comment by umireon** [comment]
I'll try it.

**Review by zkochan — APPROVED** [review]

**Comment by umireon** [comment]
@zkochan Building macos-arm64 binary on x64 Ubuntu or x64 Mac does not works. It works on M1 mac. I think the singing is related.

**Comment by umireon** [comment]
## pnpm run compile on macOS (Big Sur)

```
> Warning Cannot find module 'string_decoder/' from '/Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist'  in /Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist/pnpm.cjs
> Warning Cannot find module 'string_decoder/' from '/Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist'  in /Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist/pnpm.cjs
> Warning Cannot find module 'string_decoder/' from '/Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist'  in /Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist/pnpm.cjs
> Warning Cannot find module 'string_decoder/' from '/Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist'  in /Users/umireon/Documents/GitHub/pnpm/packages/pnpm/dist/pnpm.cjs
node:internal/child_process:415
    throw errnoException(err, 'spawn');
    ^

Error: spawn Unknown system error -86
    at ChildProcess.spawn (node:internal/child_process:415:11)
    at Object.spawn (node:child_process:698:9)
    at fabricate (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/@zkochan+pkg@0.0.0-2/node_modules/@zkochan/pkg/lib-es5/fabricator.js:55:41)
    at Object.fabricateTwice (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/@zkochan+pkg@0.0.0-2/node_modules/@zkochan/pkg/lib-es5/fabricator.js:120:5)
    at MultiStream._queue (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/@zkochan+pkg@0.0.0-2/node_modules/@zkochan/pkg/lib-es5/producer.js:266:49)
    at MultiStream._next (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/multistream@4.1.0/node_modules/multistream/index.js:86:12)
    at Meter.onEnd (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/multistream@4.1.0/node_modules/multistream/index.js:129:12)
    at Object.onceWrapper (node:events:513:28)
    at Meter.emit (node:events:394:28)
    at endReadableNT (node:internal/streams/readable:1331:12) {
  errno: -86,
  code: 'Unknown system error -86',
  syscall: 'spawn'
}

/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/execa@5.1.1/node_modules/execa/lib/error.js:60
		error = new Error(message);
          ^
Error: Command failed with exit code 1: pkg ./dist/pnpm.cjs --out-path=../artifacts/macos-arm64 --targets=node14-macos-arm64
    at makeError (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/execa@5.1.1/node_modules/execa/lib/error.js:60:11)
    at Function.module.exports.sync (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/execa@5.1.1/node_modules/execa/index.js:194:17)
    at build (/Users/umireon/Documents/GitHub/pnpm/packages/pnpm/scripts/build-artifacts.ts:5:9)
    at Object.<anonymous> (/Users/umireon/Documents/GitHub/pnpm/packages/pnpm/scripts/build-artifacts.ts:16:3)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Module.m._compile (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/ts-node@10.1.0_400a4096d34ae850b4639c830a4747bf/node_modules/ts-node/src/index.ts:1225:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1153:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/umireon/Documents/GitHub/pnpm/node_modules/.pnpm/ts-node@10.1.0_400a4096d34ae850b4639c830a4747bf/node_modules/ts-node/src/index.ts:1228:12)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
/Users/umireon/Documents/GitHub/pnpm/packages/pnpm:
 ERROR  pnpm@6.13.0-0 compile: `rimraf lib tsconfig.tsbuildinfo && tsc --build && pnpm run lint -- --fix && rimraf dist bin/nodes && pnpm run bundle && shx cp -r node-gyp-bin dist/node-gyp-bin && shx cp -r node_modules/@pnpm/tabtab/lib/scripts dist/scripts && shx cp -r node_modules/ps-list/vendor dist/vendor && ts-node ./scripts/build-artifacts.ts`
Exit status 1
 ERROR  Command failed with exit code 1.
 ERROR  Command failed with exit code 1.
```

**Comment by umireon** [comment]
Oh it works on Ubuntu.

**Pull request merged**